feat: add Dependency control automation

Totto16 opened the pull request:
[`DependencyControl`](https://github.com/TypesettingTools/DependencyControl.git) is a widely used "automation"

This PR tries to add this as an extension to this flatpak, I am new to creating Flatpak extensions, so this might take some testing 😄

Totto16 committed:
feat: add Dependency control automation

Totto16 committed:
fix: remove branch, use commit alone, where no good tag exists of the repo

zefr0x commented on org.aegisub.Aegisub.yml:
```diff
@@ -31,6 +31,13 @@ add-extensions:
     autodelete: true
+  org.aegisub.Aegisub.Automations.DependencyControl:
+    directory: automations/dependency_control
+    subdirectories: true
+    add-ld-path: lib
+    merge-dirs: share/aegisub/automation/autoload;share/aegisub/automation/include;share/aegisub/automation/demo
+    no-autodownload: true
+    autodelete: true
```
```suggestion
  org.aegisub.Aegisub.Automations.DependencyControl:
    directory: automations/dependency_control
    add-ld-path: lib
    no-autodownload: true
    autodelete: true
```

`subdirectories` and `merge-dirs` shouldn’t be needed here.

zefr0x commented on org.aegisub.Aegisub.yml:
Automations should be installed at: `${FLATPAK_DEST}/automations/dependency_control/share/aegisub/automation`
All of the install commands need to be fixed.

Totto16 replied:
Thanks for the help, I already did that, but didn't push a commit, as I have some problem with building moonscript atm 😓

zefr0x replied:
> as I have some problem with building moonscript atm 😓

I'm not familiar with lua modules or moonscript, but I might be able to help debugging each problem separately.

Totto16 replied:
I have found the error, the way LuaJIT is installed was wrong, so luajit didn't search in the correct places, when importing,
 You have to provide the prefix not only when installing, but also when building, as luajit uses the prefix for the `package.path` and `package.cpath` variables, so I fixed that, by not using  autotools and `make-install-args`.

Totto16 committed:
fix; fix installation directory for DependencyControl extension

zefr0x commented on org.aegisub.Aegisub.yml:
```diff
@@ -248,3 +255,62 @@ modules:
             x-checker-data:
               type: git
               version-scheme: semantic
+  - name: DependencyControl
```
Since flathub refused to package Aegisub plugins on separate repos.

To keep everything organised, this module with its dependencies could be moved into a separate file.

Totto16 replied:
We could and should argue, that Aegisub live at https://github.com/TypesettingTools/Aegisub nowadays, and DependencyControl is from them too: https://github.com/TypesettingTools/DependencyControl.git

zefr0x replied:
https://github.com/flathub/flathub/pull/6859

Totto16 replied:
strange 🤔  but even if they don't accept this, I'll build and use it locally myself, 😁  and for that the module works,  if the only reason to reject this extension would be that, moving it into a seperate file wouldn't be that complicated

Totto16 committed:
fix: fix luajit compilation

You have to provide the prefix not only when installing, but also when building
as luajit uses the prefix for the package.path and package.cpath variables
so doing that manually, instead of using the autotools buildsystem

Totto16 committed:
fix: fix moonscript building

Totto16 committed:
fix: use older but stable moonscript
- this uses another argparse library, so that one needs to be built

Totto16 committed:
fix: fic installation
- fix installation of files
- use luarocks to install all files of the moonscript libraray, as there are many,
too many to copy them manually

Totto16 committed:
fix: allow autodownload for org.aegisub.Aegisub.Automations.DependencyControl

Totto16 committed:
fix: install .so files for DependencyControl in a correct place

Totto16 committed:
chore: check fore files, that needs cleanup
- note: nothing needs cleanup, as all unused files are removed, all binaries like moon, moonc, luarocks are needed for Dependency control to work, but they should be stored in the bin path of the extension

Totto16 commented:
I have tested it locally, and it works as expected, if it should be refactored to a extension file like in https://github.com/flathub/flathub/pull/6859, it needs to be done in the future, and one outstanding issue still exists, the binaries are installed trough the main app, and not the extension, as this needs "moon", "moonc" and "luarocks" as binaries, to work, I am not experienced enough on how to do this with flatpak extensions tough 😓 

If this would result in the same discussion as  https://github.com/flathub/flathub/pull/6859 i am not sure how much time I am going to invest in these things, as my primary goal was, to have this for usage by myself (which i have now) and I wanted to submit it as PR, so that other can use it too, but if this doesn't get accepted into flathub either way, I have no incentive to invest too much time into it 😓  (sorry)

zefr0x commented:
bot, build

zefr0x commented:
@Totto16 please take a look at #62 and test it if you have time.
I don't use DependencyControl by myself so I can't be sure if it's working properly.
